doc/CONFIGURATION.md

```diff
@@ -0,0 +1,289 @@
+# Configuration
+
+* [Overview](#overview)
+* [Modules](#modules)
+  * [Transport](#transport)
+  * [Stream Multiplexing](#stream-multiplexing)
+  * [Connection Encryption](#connection-encryption)
+  * [Peer Discovery](#peer-discovery)
+  * [Content Routing](#content-routing)
+  * [Peer Routing](#peer-routing)
+  * [DHT](#dht)
+  * [Pubsub](#pubsub)
+* [Libp2p node](#libp2p-node)
+* [Configuration examples](#configuration-examples)
+
+## Overview
+
+Considering libp2p's modular nature, it can be composed by a set of building blocks that enable users to create any type of p2p systems without needing to implement its network layer. Taking into account that each p2p system has its own needs and requirements, libp2p can be configured accordingly.
+
+`js-libp2p` acts as the composer for this modular p2p networking stack using libp2p compatible modules as its building blocks.
+While libp2p building blocks may vary, `js-libp2p` top level API will be consistent.
+
+## Modules
+
+For getting an instance of `js-libp2p` compliant with all types of networking requirements, it is possible to specify the following building blocks:
```

Comment:
I think using `subsystems` instead of `building blocks` is a bit more descriptive.